db-engine/src/main/java/com/splicemachine/db/iapi/sql/compile/CompilerContext.java

```diff
@@ -158,6 +158,7 @@
 	boolean     DEFAULT_MULTICOLUMN_INLIST_PROBE_ON_SPARK_ENABLED = false;
 	boolean     DEFAULT_CONVERT_MULTICOLUMN_DNF_PREDICATES_TO_INLIST = true;
 	boolean     DEFAULT_DISABLE_PREDICATE_SIMPLIFICATION = false;
+	double      DEFAULT_SPLICE_SPARK_MAJOR_VERSION = 2.2;
```

Comment:
@dgomezferro Are you saying we should have an entirely new class devoted to holding the Spark version we're running?  I currently have interface getSparkMajorVersion() in the DatabaseVersion object that you mention, as spark version seems part of the database version info.  getSparkMajorVersion() retrieves a string value, picked up from a POM file setting, and its value is something like "spark2.3", which is then converted into a java property when splice is booted, with the prefix "spark" stripped off.  For ease of comparison of version numbers, this property value is read in and converted to a double value in GenericStatement.java, stored in the CompilerContext.  This way we can use '>' or '<' comparisons to check that our Spark version is at a given level or higher, e.g.

sparkVersion() >= 2.3

We can chat about this more if you like.

Reply:
Sure, we can discuss this. Yes, I was suggesting using a (simple) class for this. My reasoning is that some features might be available in Spark 2.3.2 but not int 2.3.0, so a double comparison wouldn't work. Also if we have our own patched Spark in the service a proper class could encapsulate that logic.

Reply:
@dgomezferro Added SimpleSparkVersion class.  We now set splice.spark.compile.version automatically, from POM file properties, to something like 2.2.0.cloudera2, which is used to initialize the SimpleSparkVersion class.  The cloudera2 part is ignored when comparing SimpleSparkVersion objects.  If the user sets system property splice.spark.version, and it is a valid version string, then it will override the setting in splice.spark.compile.version.  This is for the case where the cluster is running a different version of spark than the splice parcel compile-time setting for spark version.